### ISSUE TITLE
Change sort order when there are multiple online sources

### ISSUE THREAD

**Issue by escowles** [pulibrary/orangelight#931]
American theatre at https://pulsearch.princeton.edu/catalog/9783436 is a case in point. There are eight sources for the online text, starting with years 1992, 1994, or 1995. The sort is alphabetical be resource title. The results are coverage for these years: 1995, 1994, 1995, 1994, 1992, and 3 more 1994. Would users be better served by a sort based on the starting year of coverage  (earliest first), so that this list would then read 1992, 1994 (5 sources), 1995 (2 sources), perhaps subsorted by resource title? Default alphabetical order is not particularly helpful in this case.

**Comment by escowles** [comment]
Sort the online resources list by year

**Comment by tampakis** [comment]
https://getit.princeton.edu/resolve?url_ver=Z39.88-2004&ctx_ver=Z39.88-2004&ctx_enc=info:ofi/enc:UTF-8&rfr_id=info:sid/sfxit.com:opac_856&url_ctx_fmt=info:ofi/fmt:kev:mtx:ctx&sfx.ignore_date_threshold=1&rft.object_id=963018360394&svc_val_fmt=info:ofi/fmt:kev:mtx:sch_svc&

**Comment by eliotjordan** [comment]
Ticket is best addressed with the participation of @kevinreiss.

**Comment by kevinreiss** [comment]
The raw API call from our SFX system that this data comes from is: http://sfx.princeton.edu:9003/sfx_pul?url_ver=Z39.88-2004&ctx_ver=Z39.88-2004&ctx_enc=info:ofi/enc:UTF-8&rfr_id=info:sid/sfxit.com:opac_856&url_ctx_fmt=info:ofi/fmt:kev:mtx:ctx&sfx.ignore_date_threshold=1&rft.object_id=963018360394&svc_val_fmt=info:ofi/fmt:kev:mtx:sch_svc&sfx.response_type=multi_obj_xml 

That gets filtered through the SFX umlaut service adapter at https://github.com/team-umlaut/umlaut/blob/master/app/service_adaptors/sfx.rb. Whatever this parses out ultimately shows up in the HTML that umlauts Jquery plugin fetches for display in OL.

**Comment by kevinreiss** [comment]
In the past the gem maintainers tried to provide some features to sink or boost particular targets as well as collapse duplicates https://github.com/team-umlaut/umlaut/blob/master/app/service_adaptors/sfx.rb#L389-L393. In practice the library staff often objected to their use due to edge cases and inconsistent data for package titles and dates in the SFX source database.

**Comment by escowles** [comment]
@kevinreiss If I understand the way this is working correctly, OL is pulling HTML from getit.princeton.edu.  Is that something we can change the behavior of to sort the items?

**Comment by kevinreiss** [comment]
Yes, these items are pulled from the link resolver knowledgebase in SFX via the getit.princeton.edu jquery utility. We've tried to address this several times in past years but due to inconsistent date values in the knowledgebase, staff concerns around too aggressive de-duplication or the of another feature that boasts  some vendor names above others those efforts broke down. It has been some years since I looked at this problem. I think the current sort happens in this fashion:

1. Deduplicate items based on package name (at present we are only doing minimal de-duplication due to staff concerns)
2. Boast certain vendor package names to the top of the list
3. Sink certain vendor package to the bottom of the list
4. Title sort by package name
5. Some attempt at a date sort.

We could try playing around with these again, there is a https://getit-dev.princeton.edu instance we can use to test alternative approaches. Ultimately we can control what we want to send back.

**Comment by escowles** [comment]
I believe this is mostly a problem when there are a lot of items.  In that case, we might be better off just sorting by label.

**Comment by escowles** [comment]
ping @kevinreiss — can we just sort by label when there is a large (5+?) set?

**Comment by kevinreiss** [comment]
Sorry, I forgot to update this one. Before the holiday break. I activated some deduplication features that attempt to collapse items by vendor title and date range. The original example only has four links displayed now.

Yes, this is only really problem when there are lot of items, though because we have purchased so many of the sub-packages from giant vendors like ESBCO and Proquest the pattern is pretty commonly repeated for almost any long-running serial.  The displays may still seem off kilter because the committee that managed this put forth elaborate recommendations about boosting certain vendors above others because the linking on some sites work consistently better than others, which is why LexisNexis is at the bottom here: https://pulsearch.princeton.edu/catalog/9782640.
